salt/pillar/stack.py

```diff
@@ -455,9 +455,12 @@ def _process_stack_cfg(cfg, stack, minion_id, pillar):
             log.debug("YAML: basedir=%s, path=%s", basedir, path)
             # FileSystemLoader always expects unix-style paths
             unix_path = _to_unix_slashes(os.path.relpath(path, basedir))
-            obj = salt.utils.yaml.safe_load(
-                jenv.get_template(unix_path).render(stack=stack, ymlpath=path)
-            )
+            try:
+                obj = salt.utils.yaml.safe_load(
+                    jenv.get_template(unix_path).render(stack=stack, ymlpath=path)
+                )
+            except Exception as e:
+                raise Exception(f"for '{path}': {e}")
```

Comment:
```suggestion
            except Exception as e:
                raise Exception("for '{path}': {exc}".format(path=path, exc=e))
```

Maybe we also raise the same type of exception that is thrown?

Reply:
Sounds like a good idea

Reply:
I think the main use case I was going for here was a YAML parse error. Since you can't reliably modify an arbitrary exception's message, I switched to just capturing the YAML exception and reconstructing that because its constructor signature is known.

If there are any other common specific exceptions that you think would also occur here lmk.

Reply:
Maybe it should return this exception as a chain from the original so that the stack trace information is not lost:

```python
raise YamlException("for '{path}': {exc}".format(path=path, exc=e) from e
```